src/replication/adapter/helmhub/client.go

```diff
@@ -0,0 +1,97 @@
+package helmhub
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/goharbor/harbor/src/replication/model"
+	"github.com/goharbor/harbor/src/replication/util"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+)
+
+// ErrHTTPNotFound defines the return error when receiving 404 response code
+var ErrHTTPNotFound = errors.New("Not Found")
+
+// Client is a client to interact with HelmHub
+type Client struct {
+	client *http.Client
+}
+
+// NewClient creates a new HelmHub client.
+func NewClient(registry *model.Registry) *Client {
+	return &Client{
+		client: &http.Client{
+			Transport: util.GetHTTPTransport(false),
```

Comment:
Should read the `insecure` property of `registry` to set the parameter to `GetHTTPTransport`

Reply:
the helm hub URL is fixed. and it is from offical. so secure should be checked. also refering to dockerhub.